float32 velPercentage
string keyStroke 